zad5/app/controllers/artists_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ArtistsController < ApplicationController
+  before_action :set_artists_collection, only: %i[new edit create update]
+
+  def show
+    @artist = Artist.find(params[:id])
+  end
+
+  def index
+  	@artists = Artist.all
+  end
+
+  def new
+  	@artist = Artist.new
+  end
+
+  def create
+  	@artist = Artist.new(artist_params)
+  	if @artist.save
+  	  redirect_to artists_url, notice: 'Artist has been saved'
+  	else
+  	  render 'new'
+  	end
+  end
+
+  def edit
+    @artist = Artist.find(params[:id])
+    render 'edit'
+  end
+
+  def update
+    if @artist.update(artist_params)
+      redirect_to artists_url, notice: 'Artist has been updated'
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    Artist.destroy(params[:id])
+    redirect_to artists_url, notice: 'Artist has been wiped out'
+  end
+
+  private
+
+  def set_artists_collection
```

Comment:
nazwa może być myląca, chcemy jednego artystę, a nie kolecję
dodatkowo z `find_by_id(id)` może być problem tutaj bo:
1. ta składnia jest trochę deprecated i narzędzia typu rubocop by tutaj krzyczały, aby przepisać to na `find_by(id: id)`
2. jeśli nie znajdziemy artysty chcemy, aby strona zwróciła nam 404, co się dzieje w przypadku zwykłego finda (`ActiveRecord::RecordNotFound`)(`find(id)`), tymczasem `find_by_id` / `find_by(id: id)` zwróci nam nila